docs/08-how-to-submit-a-proof/01-javascript-example.md

```diff
@@ -0,0 +1,179 @@
+
+3. **Verify Installation**:
+   - Open a terminal or command prompt.
+   - Run the following commands to verify that Node.js and npm are installed correctly:
+     ```shell
+     node -v
+     npm -v
+     ```
+   - You should see the version numbers of Node.js and npm.
+
+Now you are ready to follow the instructions below to submit a proof.
+
+### Initialize Your Project
+
+1. Create a new directory for your project, navigate into it:
+
+    ```shell
+    mkdir my-zkverify-project
+    cd my-zkverify-project
+    ```
+2. Initialize a new npm project:
+
+    ```shell
+    npm init -y
+    ```
+3. Install the required packages:
+
+    ```shell
+    npm install @polkadot/api @polkadot/keyring dotenv
+    ```
+4. Create a `.env` file:
+
+    ```shell
+    echo -e "WEBSOCKET=wss://testnet-rpc.zkverify.io\nPRIVATE_KEY=\"below giant big little small royal three two one yes test tag\"" > .env
+    ```
+5. Update the `.env` file values:
+
+   - WEBSOCKET: The zkVerify Websocket url
+   - PRIVATE_KEY: Your testnet seedphrase enclosed in quotation marks
+
+### Example Script
+
+Create a file named `submitProof.js` and paste the following script into it:
+
+```javascript
+require('dotenv').config();
+const { ApiPromise, WsProvider } = require('@polkadot/api');
+const { Keyring } = require('@polkadot/keyring');
+
+const proofs = {
+   fflonk: {
+      pallet: 'settlementFFlonkPallet'
+   },
+   boojum: {
+      pallet: 'settlementZksyncPallet'
+   }
+};
+
+if (!process.env.WEBSOCKET || !process.env.PRIVATE_KEY) {
+   throw new Error('Required environment variables WEBSOCKET or PRIVATE_KEY are not set.');
+}
+
+const proofType = process.argv[2];
+const proofHash = process.argv[3];
+
+if (!proofType || !proofHash) {
+   throw new Error('Please provide both proof type and proof hash as command line arguments.');
+}
+
+if (!proofs[proofType]) {
+   throw new Error('Invalid proof type. Valid proof types are: fflonk, boojum.');
+}
+
+(async () => {
+   let api;
+   let provider;
+   try {
+      provider = new WsProvider(process.env.WEBSOCKET);
+
+      const timeout = new Promise((_, reject) =>
+              setTimeout(() => reject(new Error('Failed to connect to the WebSocket URL. Please check the WEBSOCKET environment variable.')), 3000)
+      );
+
+      api = await Promise.race([ApiPromise.create({ provider }), timeout]);
+
+      const keyring = new Keyring({ type: 'sr25519' });
+      const account = keyring.addFromUri(process.env.PRIVATE_KEY);
+
+      const submitProof = api.tx[proofs[proofType].pallet].submitProof(proofHash);
+
+      console.log(`Submitting ${proofType} proof...`);
+      await submitProof.signAndSend(account, ({ status, dispatchError }) => {
+         if (status.isInBlock) {
+            console.log(`Transaction included at blockHash: https://testnet-explorer.zkverify.io/block/${status.asInBlock}`);
+         }
+         if (status.isFinalized) {
+            if (dispatchError) {
```

Comment:
I think you can check errors immediately, no need to wait for isFinalized status